File: LeetCode/Kai/43MultiplyStrings.java
public class Solution {
    public String multiply(String num1, String num2) {
        int len1 = num1.length();
        int len2 = num2.length();
        int[] output = new int[len1+len2];     //result will be no more than len1+len2 digits
        for(int i=len1-1; i>=0; i--) {
            for(int j=len2-1; j>=0; j--) {
                int index = len1+len2-i-j-2;
                output[index] += (num1.charAt(i)-'0') * (num2.charAt(j)-'0');
                output[index+1] += output[index]/10;
                output[index] = output[index]%10;
            }
        }
        
        StringBuilder sb = new StringBuilder();
        for(int i=output.length-1; i>0; i--) {
            if(sb.length()==0 && output[i]==0) continue;
            sb.append(output[i]);
        }
        sb.append(output[0]);
        return sb.toString();
    }
    
}

File: LeetCode/Peng/43MultiplyStrings.cpp
/**
 * Just add the carry to the next digit, finally just check the highest
 * digit.
 **/ 
class Solution {
public:
    string multiply(string num1, string num2) {
        int l1 = num1.size();
        int l2 = num2.size();
        if (!l1 || !l2 || num1 == "0" || num2 == "0") return "0";
        
        string result = "";
        vector<int> digits(l1 + l2, 0);
        for (int i = l1 - 1; i >= 0; i--) {
            int a = num1.at(i) - '0';
            int pos = l1 - i - 1;
            for (int j = l2 - 1; j >= 0; j--, pos++) {
                int b = num2.at(j) - '0';
                int temp = a * b + digits.at(pos);
                digits.at(pos) = temp % 10;
                digits.at(pos + 1) += temp / 10;
            }
        }
        
        if (digits.at(l1 + l2 - 1)) {
            result += to_string(digits.at(l1 + l2 - 1));
        }
        for (int i = digits.size() - 2; i >= 0; i--) {
            result += to_string(digits.at(i));
        }
        return result;
    }
};